lxc/completion.go

```diff
@@ -377,12 +375,9 @@ func (g *cmdGlobal) cmpInstanceSetKeys(instanceName string) ([]string, cobra.She
 	for configKey := range instance.Config {
 		// We only want to return the intersection between allInstanceConfigKeys and configKeys to avoid returning the full instance config.
 		_, exists := keySet[configKey]
-		if exists {
-			if shared.StringHasPrefix(configKey, instancetype.ConfigKeyPrefixesAny...) {
-				configKeys = append(configKeys, configKey)
-			} else if instanceType == string(api.InstanceTypeContainer) && shared.StringHasPrefix(configKey, instancetype.ConfigKeyPrefixesContainer...) {
-				configKeys = append(configKeys, configKey)
-			}
+		knownPrefixes := append(instancetype.ConfigKeyPrefixesAny, instancetype.ConfigKeyPrefixesContainer...)
```

Comment:
Separately, so following on from yesterday's comment, it seems like you're already depending on build-time config keys from `instancetype` package, which could differ from the keys supported by the target server (as lxc client version may differ from server version). Here it looks to me like this could end up with a situation where valid keys supported by the server are not suggested by the client, or vice versa.


However, if you've already gone this route here, then why did you go the route of using the metadata `Condition` field in the other PR? It seems to me we should pick one approach for gathering supported keys, either a build-time list (and accept the drawbacks that it may mismatch what is supported) or use the metadata API.

Mixing the approach seems confusing to me.